step-functions/nino_issue_credential.asl.json

```diff
@@ -1,32 +1,63 @@
 {
   "Comment": "A description of my state machine",
-  "StartAt": "Query Session Item",
+  "StartAt": "Fetch Session ID for AccessToken",
   "States": {
-    "Query Session Item": {
+    "Fetch Session ID for AccessToken": {
       "Type": "Task",
       "Parameters": {
         "TableName": "session-${CommonStackName}",
-        "FilterExpression": "accessToken = :value",
+        "IndexName": "access-token-index",
+        "KeyConditionExpression": "accessToken = :value",
         "ExpressionAttributeValues": {
           ":value": {
             "S.$": "$.bearerToken"
           }
         }
       },
-      "Resource": "arn:aws:states:::aws-sdk:dynamodb:scan",
-      "ResultPath": "$.querySessionResult",
-      "Next": "Bearer Token Valid?"
+      "Resource": "arn:aws:states:::aws-sdk:dynamodb:query",
+      "Next": "Did we find a session for the given Bearer Token?",
```

Comment:
Could we use consistent language? We say "... for access token" above and "... for given bearer token" here